src/test/java/cz/muni/fi/pa165/TrainerFightDaoTest.java

```diff
@@ -0,0 +1,275 @@
+    void testSaveTrainerFightWithNullTrainerAndNullGym() {
+        trainerFight = new TrainerFight();
+        trainerFight.setTargetGym(null);
+        trainerFight.setChallenger(null);
+        trainerFight.setWasChallengerSuccessful(true);
+        trainerFightDao.save(trainerFight);
+    }
+
+
+    /** delete(...) tests */
+
+    @Test(expectedExceptions = {DataAccessException.class})
+    void testDeleteNullTrainerFight() {
+        trainerFight = null;
+        assertThat(trainerFight, is(nullValue(TrainerFight.class)));
+        trainerFightDao.delete(trainerFight);
+    }
+
+    @Test(expectedExceptions = {DataAccessException.class})
+    void testDeleteWithNonexistentId() {
+        trainerFightDao.delete(99999999999L);
+    }
+
+    @Test
+    void testDeleteByPassingInstance() {
+        trainerFight = randomTrainerFight(5);
+        assertThat(trainerFightDao.count(), is(equalTo(0L)));
+        trainerFightDao.save(trainerFight);
+        assertThat(trainerFightDao.count(), is(equalTo(1L)));
+        assertThat("trainerFight.getId() == null", trainerFight.getId(), is(notNullValue(Long.class)));
+        trainerFightDao.delete(trainerFight);
+        assertThat(trainerFightDao.count(), is(equalTo(0L)));
+    }
+
+    @Test
+    void testDeleteByPassingId() {
+        trainerFight = randomTrainerFight(5);
+        assertThat(trainerFightDao.count(), is(equalTo(0L)));
+        trainerFightDao.save(trainerFight);
+        assertThat(trainerFightDao.count(), is(equalTo(1L)));
+        assertThat("trainerFight.getId() == null", trainerFight.getId(), is(notNullValue(Long.class)));
+        trainerFightDao.delete(trainerFight.getId());
+        assertThat(trainerFightDao.count(), is(equalTo(0L)));
+    }
+
+    @Test
+    void testDeleteMultipleInstances() {
+        List<TrainerFight> trainerFights = randomTrainerFights(5);
+        assertThat(trainerFightDao.count(), is(equalTo(0L)));
+        trainerFightDao.save(trainerFights);
+        assertThat(trainerFightDao.count(), is(equalTo(5L)));
+        trainerFightDao.delete(trainerFights);
+        assertThat(trainerFightDao.count(), is(equalTo(0L)));
+    }
+
+    /** find(...) tests */
+
+    @Test
+    void testFindByPassingId() {
+        trainerFight = randomTrainerFight(5);
+        trainerFightDao.save(trainerFight);
+        assertThat(trainerFight.getId(), is(notNullValue()));
+        TrainerFight found = trainerFightDao.findOne(trainerFight.getId());
+        assertThat(found, is(equalTo(trainerFight)));
+    }
+
+    @Test
+    void testFindByPassingExample() {
+        trainerFight = randomTrainerFight(5);
+        trainerFightDao.save(trainerFight);
+        assertThat(trainerFight.getId(), is(notNullValue()));
+        TrainerFight trainerFightExample = new TrainerFight();
+        trainerFightExample.setWasChallengerSuccessful(true);
+        TrainerFight found = trainerFightDao.findOne(Example.of(trainerFightExample));
```

Comment:
Zas: Example funguje tak, ze vsetky fieldy co nastavis, tak on da do SELECT query, tj, ak to `randomTrainerFight()` nastavi `wasChalelngerSuccessFul` na `false` a ty potom das example, kde to je `true` tak nic nenajde.